protocol.go

```diff
@@ -81,10 +82,11 @@ func ProcedureFromURL(url *url.URL) (string, bool) {
 		return url.Path, false
 	}
 	procedure := path[penultimate:]
-	if len(procedure) < 4 { // two slashes + service + method
+	// Ensure that the service and method are non-empty.
+	if ultimate == len(path)-1 || penultimate == ultimate-1 {
 		return url.Path, false
 	}
-	return procedure, true
+	return procedure, false
```

Comment:
This change looks like an accident. This was the one case to return true, right?